[MRG+1] Simplifiy example

hdmetor opened the pull request:
Fix for https://github.com/scikit-learn/scikit-learn/issues/5720

hdmetor committed:
Using rain_test_split and StandardScaler instead of reimplementing them

agramfort commented:
LGTM

GaelVaroquaux commented on examples/applications/plot_prediction_latency.py:
```diff
@@ -93,27 +96,16 @@ def generate_dataset(n_train, n_test, n_features, noise=0.1, verbose=False):
+    X = StandardScaler().fit_transform(X)
+    y = StandardScaler().fit_transform(y)
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=n_train)
```
If we want things to be reproducible, we should also set the random_state on train_test_split.

TomDLT commented on examples/applications/plot_prediction_latency.py:
```diff
@@ -93,27 +96,17 @@ def generate_dataset(n_train, n_test, n_features, noise=0.1, verbose=False):
+    y = StandardScaler().fit_transform(y)
+
+    random_seed= 13
+    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=n_train, random_state=random_seed)
```
this line is not PEP8 (above 79 characters by line)

TomDLT commented on examples/applications/plot_prediction_latency.py:
```diff
@@ -93,27 +96,17 @@ def generate_dataset(n_train, n_test, n_features, noise=0.1, verbose=False):
+    X = StandardScaler().fit_transform(X)
+    y = StandardScaler().fit_transform(y)
+
+    random_seed= 13
```
please add a space before the `=`

TomDLT commented:
apart from PEP8,
LGTM

hdmetor commented:
Also please not that I get a lot of the following warnings on Python 3.5.0

    /usr/local/lib/python3.5/site-packages/sklearn/base.py:175: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() instead
    args, varargs, kw, default = inspect.getargspec(init)

hdmetor commented:
After updating to `scikit-learn` 0.17 and `matplotlib` version 1.5.0 I don't get the same error anymore. Instead I get:

    /usr/local/lib/python3.5/site-packages/sklearn/utils/validation.py:386: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and willraise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.

Should that be resolved in a different ticket since it's a different issue?

GaelVaroquaux commented:
LGTM. +1 for merge.

To make it possible to be efficient when reviewing pull request, it would be great if people could 1) give a human-readable name to the PR, 2) use the [MRG] for PRs ready to merge, and [MRG+1] when one core dev has given his +1.

agramfort commented on examples/applications/plot_prediction_latency.py:
```diff
@@ -93,27 +96,17 @@ def generate_dataset(n_train, n_test, n_features, noise=0.1, verbose=False):
-    y_test = (y_test - mean) / std
+
+    X = StandardScaler().fit_transform(X)
+    y = StandardScaler().fit_transform(y)
```
wait a minute. This is buggy.

before the scaling was done using train data even for the test. Now you scale globally which is wrong.

the proper way is to use a pipeline

GaelVaroquaux replied:
> wait a minute. This is buggy.

> before the scaling was done using train data even for the test. Now you scale
> globally which is wrong.

Good catch.

> the proper way is to use a pipeline

That cannot scale y :(.

hdmetor commented:
@GaelVaroquaux Thanks for the hints about naming the PR. I think it might be useful to add this to 
the [contributors guidelines](https://github.com/scikit-learn/scikit-learn/wiki/Contributors-Guide)? 
I am not sure if by 'people' you mean contributors or owners.

Since we can't use a Pipeline I have added two manual scalers to fix the bug

agramfort commented:
now really LGTM

hdmetor commented:
For some reason it fails on CircleCi. Is there a way of forcing the check again?

TomDLT commented:
Could you rebase ?

hdmetor commented:
Rebased!

TomDLT commented:
You seem to have a problem :)
You should have only your commit in the pull-request, and ideally only one (squashed) commit.
Do you need help?

hdmetor commented:
I have rebased and squashed everything in (almost) one commit.For some reason it is showing a lot of them. Any insight would be appreciated. Hopefully tests will pass this time :)

TomDLT commented:
```python
### squash in one commit:
git checkout bugfix/5720
git rebase -i HEAD~9
# replace all the 'pick' by 'f', except the first one
# and make sure these are only your commits, starting with
# "AUsing rain_test_split and StandardScaler instead of reimplementing them"

### note the hash code of your commit with git log
# it should be a 40 characters string with numbers and letters,
# for example something like 2bbeea1b8a5cef12830927bdcdd41324c7312281
git log

### make a backup of your work
git checkout -b backup_branch

### delete master and refetch a clean one:
git branch -D master
git fetch upstream master:master

### delete your branch and create a new one with the same name, from master
git checkout master
git branch -D  bugfix/5720
git checkout -b  bugfix/5720

### cherry-pick your commit using the hash code
git cherry-pick 2bbeea1b8a5cef12830927bdcdd41324c7312281q
# if there are some conflicts, solve them, add the files, then continue
git add this_file_with_conflicts_i_just_solved
git cherry-pick --continue

### verify you have only one commit above current master, then force push
git push -f origin bugfix/5720
```

hdmetor commented:
Thanks for your help, I was missing the cherry-picking part. Will do asap

At least now tests are passing :)

amueller commented:
81 commits?

hdmetor commented:
I rebased, and I messed up. I need to redo the process and cherry-picking so I can have only one commit

TomDLT commented:
Just to be clear, cherry-picking is done here to recover your mess, but the standard process for squashing 3 commits in 1, and for rebasing on top of master is just:
```python
# update master
git checkout master
git pull upstream master
# squash
git checkout my_branch
git rebase -i HEAD~3
# rebase on top of master
git rebase master
```

hdmetor commented:
@TomDLT Thanks a lot the the details instructions. Now the PR is clean. There is a failure, which does not seem related to my code changes:

    downloading Cal. housing from http://lib.stat.cmu.edu/modules.php?op=modload&name=Downloads&file=index&req=getit&lid=83 to /home/ubuntu/scikit_learn_data
    Failed downloading california housing data.
    ____________________Traceback (most recent call last):
      File "/home/ubuntu/scikit-learn/doc/sphinxext/gen_rst.py", line 864, in generate_file_rst
        execfile(os.path.basename(src_file), my_globals)
      File "plot_partial_dependence.py", line 67, in <module>
        sys.exit()
    SystemExit

TomDLT commented:
Thanks @hdmetor, don't worry about the CircleCI failure.

However, I just realized you add a lot of warnings in the example.
Could you fix them?
- you should give 2D arrays to `StandardScaler`:
```python
    y_scaler = StandardScaler()
    y_train = y_scaler.fit_transform(y_train[:, None])[:, 0]
    y_test = y_scaler.transform(y_test[:, None])[:, 0]
```
- you should also import `train_test_split` from `sklearn.model_selection` instead of `sklearn.cross_validation`

hdmetor commented:
@TomDLT thanks for your help, warnings went away.
Let me know if you want me to squash in one commit again. This time should be less painful

TomDLT commented:
> Let me know if you want me to squash in one commit again. This time should be less painful

Please do, thanks !

hdmetor commented:
Ok this time it worked without me screwing up!

TomDLT commented:
Merged, thanks @hdmetor !!
